packages/transport/src/modules/accounts/AccountController.ts

```diff
@@ -235,11 +236,29 @@ export class AccountController {
     public async syncDatawallet(force = false): Promise<void> {
         if (!force && !this.autoSync) return;
 
-        await this.synchronization.sync("OnlyDatawallet");
+        const changedItems = await this.synchronization.sync("OnlyDatawallet");
+        await this.triggerEventsForChangedItems(changedItems);
     }
 
     public async syncEverything(): Promise<ChangedItems> {
-        return await this.synchronization.sync("Everything");
+        const changedItems = await this.synchronization.sync("Everything");
+        await this.triggerEventsForChangedItems(changedItems);
+
+        return changedItems;
+    }
+
+    private async triggerEventsForChangedItems(changedItems: ChangedItems) {
+        const changedIdentityDeletionProcessIds = changedItems.changedObjectIdentifiersDuringDatawalletSync.filter((x) => x.startsWith("IDP"));
+        for (const id of changedIdentityDeletionProcessIds) {
+            const changedIdentityDeletionProcesses = await this.identityDeletionProcess.getIdentityDeletionProcess(id);
```

Comment:
Switch the variable name to singular `changedIdentityDeletionProcess` please as at most one IdentityDeletionProcess can be returned by the method call `await this.identityDeletionProcess.getIdentityDeletionProcess(id)`.